mixin/lib/civimix-schema@5/src/SqlGenerator.php

```diff
@@ -115,7 +115,19 @@ private function getTableDefinition(array $entity): array {
       if (!empty($field['primary_key'])) {
         $primaryKeys[] = "`$fieldName`";
       }
-      $definition[] = "`$fieldName` " . self::generateFieldSql($field);
+      // if we have localizable columns and we are in multilingual mode generate columns in the multilingual format.
+      if (!Civi\Core\Container::isContainerBooted()) {
+        $definition[] = "`$fieldName` " . self::generateFieldSql($field);
```

Comment:
So what's the implication here? We create the schema incorrectly sometimes?

Reply:
Potentially yes 

So if you install the extension I mentioned in the description you get a column title even if the site is in multilingual. This breaks if you try and run any DB queries against the table in a multilingual site because it will try and reference the multilingual view https://github.com/civicrm/civicrm-core/blob/master/CRM/Core/I18n/Schema.php#L341 because it will rewrite any entity's table where that entity has a column that is marked as localizable https://github.com/civicrm/civicrm-core/blob/master/CRM/Core/I18n/SchemaStructure.php#L35 and the whole I18n schema assumes that the table has columns like <fieldName>_<locale> for all locales enabled and then it creates views of that table for each locale so the column names are just <fieldName>

Reply:
see also https://github.com/civicrm/civicrm-core/blob/master/CRM/Core/I18n/Schema.php#L459